### DISCUSSION TITLE
req.auth in NextJS middleware showing default user object

### DISCUSSION THREAD

**Discussion by DennisSpek** [nextauthjs/next-auth#11496] [Help]
Hi all,

I recently implemented AuthJS to handle authentication in my app. I modified both the JWT and Session callbacks within my `auth.ts` and so far, both the sessions from `await auth()` used in my page and `useSession()` used in my client component show the updated user objects.

The issue comes when I log the  session used in my Middleware. The session object in my Middleware is the default one whilest I would expect all sessions to be the same objects.

My `auth.ts` is as follows:

```
return {
    adapter: PostgresAdapter(pool),
    session: { strategy: "jwt" },
    ...authConfig,
    callbacks: {
      jwt({token, user, account, profile, isNewUser}) {
        if (user) { // User is available during sign-in
          token.id = user.id
          token.isNewUser = isNewUser
        }

        return token
      },
      session({ session, token }) {

        console.log("token", token);

        session.user.id = token.id;
        session.user.address = "123 Fake St.";
        session.isNewUser = token.isNewUser;

        return session
      },
    },
  }
```
My Middleware is as follows:

```
import authConfig from "./auth.config"
import NextAuth from "next-auth"

const { auth } = NextAuth(authConfig)
export default auth(async function middleware(req: NextRequest) {
  // Middleware to redirect unauthenticated users to the login page/default '/' page
    if (!req.auth && req.nextUrl.pathname !== "/") {
      const newUrl = new URL("/", req.nextUrl.origin)
      return Response.redirect(newUrl)
    }

    console.log("req.auth", req.auth);
})

export const config = {
  matcher: [
    '/((?!api|_next/static|_next/image|favicon.ico|^/$).*)',
  ],
}
``` 

The `console.log` returns the following object, which is the default object:
```
{
  user: { name: 'Test', email: 'Test@test.com', image: null },
  expires: '2024-08-31T16:12:59.203Z'
}
```

Any reason why the object returned within my Middleware, does not include the user ID and isNewUser boolean?

**Comment by k-taro56** [comment]
I am not very sure, but I believe that only the `authorize` callback is called in the middleware.

https://authjs.dev/reference/nextjs#in-middleware

**Comment by tachyon322** [comment]
Did you solve this problem?? It can be so helpful for me now
